openpgp/internal/encoding/octet_stream.go

```diff
@@ -0,0 +1,61 @@
+package encoding
+
+import (
+	"io"
+
+	"github.com/ProtonMail/go-crypto/openpgp/errors"
+)
+
+type OctetString struct {
+	length	uint16
+	data []byte
+}
+
+func NewOctetString(data []byte) (*OctetString, error) {
+	byteLength := len(data)
+	if byteLength > 65535 {
+		return nil, errors.InvalidArgumentError("Data too long")
+	}
+
+	trimmedByteLength := uint16(byteLength)
+	return &OctetString{trimmedByteLength, data}, nil
+}
+
+func (stream *OctetString) Bytes() []byte {
+	return stream.data
+}
+
+func (stream *OctetString) BitLength() uint16 {
+	return stream.length * 8
+}
+
+func (stream *OctetString) EncodedBytes() []byte {
+	buffer := make([]byte, stream.length + 2)
+
+	encodedLength := [2]byte{
+		uint8((stream.length >> 8) & 255),
+		uint8(stream.length & 255),
+	}
+
+	copy(buffer, encodedLength[:])
+	copy(buffer[2:], stream.data)
+
+	return buffer
+}
+
+func (stream *OctetString) EncodedLength() uint16 {
+	return stream.length + 2
+}
+
+func (stream *OctetString) ReadFrom(r io.Reader) (int64, error) {
+	var lengthBytes [2]byte
+	if _, err := r.Read(lengthBytes[:]); err != nil {
```

Comment:
`Read` doesn't guarantee that the slice is filled, so you have to use `io.ReadFull` here (same below)